utils/clients/busapps_ingestion_client.py

```diff
@@ -0,0 +1,125 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Client class for ingesting BusApps data via SOAP
+"""
+from suds import Client, WebFault
+from utils.clients.abstract_client import AbstractClient
+from utils.clients.connection_exception import ConnectionException
+from utils.test_settings import BUSAPPS_SETTINGS
+
+
+class BusAppsIngestionClient(AbstractClient):
+    """
+    Class for client to ingest BusApps data via SOAP
+    """
+    def __init__(self, credentials=None):
+        if not credentials:
+            credentials = BUSAPPS_SETTINGS
+        super(BusAppsIngestionClient, self).__init__(credentials)
+        self._uows_client = None
+        self._scheduler_client = None
+        self._session_id = None
+        # TODO: Note - I've used an error dictionary (for reuse and more granular testing)  # pylint:disable=fixme,line-too-long
+        self._errors = {
+            "invalid_uows_client": TypeError("The UOWS Client does not exist"
+                                             "or has not been initialised properly"),
+            "invalid_scheduler_client": TypeError("The Scheduler Client does not exist"
+                                                  "or has not been initialised properly"),
+            "invalid_session_id": ConnectionException("BusApps Ingestion")
+        }
+
+    def create_uows_client(self):
+        """ Creates a User Office Web Service Client is one does not exist """
+        if self._uows_client is None:
+            self._uows_client = Client(self.credentials.uows_url)
```

Comment:
to be verbose and to avoid confusion, I wonder if we would be better changing the import to:
`import suds` and then referencing these clients as `suds.Client`. I only say this Autoreduction has it's own understanding of a `Client` and seeing it written here it's not immediately obvious what the client is